coreplugins/cesiumion/public/components/UploadDialog.jsx

```diff
@@ -0,0 +1,222 @@
+import React, { Component, Fragment } from "react";
+
+import FormDialog from "../../../../app/static/app/js/components/FormDialog";
+
+import BootstrapField from "./BootstrapField";
+import { ImplicitIonFetcher as IonFetcher } from "./Fetcher";
+import { AssetType, SourceType } from "../defaults";
+import "./UploadDialog.scss";
+
+export default class UploadDialog extends Component {
+	static AssetSourceType = {
+		[AssetType.ORTHOPHOTO]: SourceType.RASTER_IMAGERY,
+		[AssetType.TERRAIN_MODEL]: SourceType.RASTER_TERRAIN,
+		[AssetType.SURFACE_MODEL]: SourceType.RASTER_TERRAIN,
+		[AssetType.POINTCLOUD]: SourceType.POINTCLOUD,
+		[AssetType.TEXTURED_MODEL]: SourceType.CAPTURE
+	};
+
+	static defaultProps = {
+		show: true,
+		asset: null,
+		loading: false,
+		initialValues: {
+			name: "",
+			description: "",
+			attribution: "",
+			options: {
+				baseTerrainId: "",
+				textureFormat: false
+			}
+		}
+	};
+
+	constructor(props) {
+		super(props);
+
+		this.mergedInitialValues = {
+			...UploadDialog.defaultProps.initialValues,
+			...this.props.initialValues
+		};
+
+		this.state = {
+			title : props.title,
+			...this.mergedInitialValues
+		}
+	}
+
+    show(){
+        this.dialog.show();
+    }
+
+    handleChange = (e) => {
+        const { value, name } = e.target;
+		
+		if (name === "options.textureFormat")
+		{
+			let options = {...this.state.options};
+			options["textureFormat"] = value === "Yes";
+			this.setState({ options });
+		}
+		else if (name === "options.baseTerrainId")
+		{
+			let options = {...this.state.options};
+			options["baseTerrainId"] = value;
+			this.setState({ options });
+		}
+		else
+		{
+			this.setState({ [name]: value });
+		}
+    }
+
+	handleError = msg => error => {
+		this.props.onError(msg);
+		console.error(error);
+	};
+
+	onSubmit = values => {
+		const { asset, onSubmit } = this.props;
+		values = JSON.parse(JSON.stringify(this.state));
```

Comment:
Why do we need to do this?

Reply:
This is used to make the JSON/map before the back-end can start working on the upload as it is part of the headers/keys cesium ion needs to accept the uploading asset. (It is required to convert the `this.state` to a JSON acceptable header for the upload)